dask/array/core.py

```diff
@@ -465,6 +469,8 @@ def __len__(self):
 
     @property
     def dtype(self):
+        if self._dtype is not None:
+            return self._dtype
         if self.shape:
             return self[(0,) * self.ndim].compute().dtype
```

Comment:
Fixed.  Note though that I think we've covered most of the operations.  I haven't yet done `ones`, `zeros`, and `da.random`, but I think that this PR covers most things.